scripts/debug_tools/failure_lib.py

```diff
@@ -0,0 +1,48 @@
+import json
+
+
+def findPath(string, path_begin):
+    """ Find one path in string.
+    @return: the pair of the begin and end pos
+    """
+    path_end = string.find(' ', path_begin)
+    if path_end == -1:
+        path_end = len(string)
+    path = string[path_begin:path_end]
+    return (path, path_end)
+
+
+def changePaths(string, pathModifierFun):
+    """ Scan through the string and possibly replace all found paths.
+    @return: The modified string
+    """
+    result = ''
+    i = 0
+    while (i < len(string)):
+        # Everything is a path wich starts with a '/' and there is a whitespace
+        # after that
+        if string[i] == '/':
+            (path, path_end) = findPath(string, i)
+            path = pathModifierFun(path)
+            # print(path, end='')
+            result += path
+            i = path_end - 1
+        else:
+            # print(string[i], end='')
+            result += string[i]
+        i = i + 1
+    return result
+
+
+class IncludePathModifier:
+    def __init__(self, sources_root):
+        self.sources_root = sources_root
+
+    def __call__(self, path):
+        return ''.join((self.sources_root, path))
```

Comment:
Isn't it better to use os.path.join here?

Reply:
Yes, we could use os.path.join here.
But, currently these debug tools can handle only POSIX paths, they won't work with windows paths.
This is because currently we handle a path as a list of consecutive characters from `/` to a space character.

Reply:
I prefer not to build technical debt. We cannot know whether windows will be a supported platform in the future and while we do not test things on windows, we should not introduce something on purpose that might cause problems there.

Reply:
Don't get me wrong, i am not against using os.path.join, and I already changed that. But I am against supporting windows paths now. Atm that won't help us at all in debugging, but would make these scripts more harder to maintain and write.

Reply:
I do not see why is it harder to maintain. I think it does not really have any downsides but it might help us in the future. Also, I find it more readable because that makes it clear it is an operation on paths and not just random strings without looking at the arguments.

Reply:
I think there is a misunderstanding here.
I agree to use os.path.join, and the code is already changed in that way.

Reply:
Note, that in the while loop here https://github.com/Ericsson/codechecker/pull/1113/files#diff-3e4a613d13e84f2ba4cc26a770517e5fR22 we are still bound to POSIX paths and I don't really know how could we easily handle windows paths as well. I am open to suggestions, but I'd like to avoid ugly regexes and complex implementation.

Reply:
The crux of the problem is how to detect for a substring if that is a path either in windows or in unix?

Reply:
There is `os.path.commonpath`, `os.path.commonprefix`, `os.path.split`. Does any of that help to make this cleaner?